src/components/TaskConfirmationModal/TaskConfirmationModal.jsx

```diff
@@ -216,25 +221,73 @@ export class TaskConfirmationModal extends Component {
-                     </div>
-
-                     <div>
-                       <textarea
-                         ref={this.commentInputRef}
-                         className="mr-input mr-text-white mr-placeholder-medium mr-bg-grey-lighter-10 mr-border-none mr-shadow-inner mr-p-3 mr-mt-1"
-                         rows={2}
-                         cols="1"
-                         value={this.props.osmComment}
-                        onChange={e => this.props.setOSMComment(e.target.value)}
-                       />
+                     <div className="mr-border mr-border-gray-300 mr-shadow-md mr-p-4 mr-mt-4">
+                       <p className="mr-mb-2 mr-text-white mr-text-md mr-font-bold mr-bg-blue-600 mr-rounded">
+                         <FormattedMessage
+                           {...messages.osmUploadNotice }
+                         />
+                       </p>
+                       <div className="mr-text-base mr-mt-2 mr-text-yellow">
+                         <FormattedMessage {...messages.osmCommentHeader} />
+                       </div>
+
+                       <div>
+                         <textarea
+                           ref={this.commentInputRef}
+                           className="mr-input mr-text-white mr-placeholder-medium mr-bg-grey-lighter-10 mr-border-none mr-shadow-inner mr-p-3 mr-my-1"
+                           rows={2}
+                           value={this.props.osmComment}
+                           onChange={e => this.props.setOSMComment(e.target.value)}
+                         />
+                       </div>
+
+                       <div className="mr-bg-blue-dark shadow-md">
+                         <div className="mr-flex mr-justify-between">
+                           <ul className="mr-w-1/3 mr-px-2 mr-border mr-border-gray-300 mr-shadow-sm">
+                             <li className="mr-font-bold mr-py-1 mr-text-sm">Tag Name</li>
+                             {_get(this.props, 'tagDiffs[0]') && Object.keys(_get(this.props, 'tagDiffs[0]')).map(tagName => {
+                               const tagChange = _get(this.props, 'tagDiffs[0]')[tagName];
+                               if (['changed', 'removed', 'added'].includes(tagChange.status)) {
+                                 return (
+                                   <li key={tagName} className={classNames('mr-mb-2  mr-rounded', {
+                                     'mr-text-orange': tagChange.status === 'changed',
+                                     'mr-text-lavender-rose': tagChange.status === 'removed',
+                                     'mr-text-picton-blue': tagChange.status === 'added',
+                                   })}>
+                                     <strong className="mr-text-sm">{tagName}</strong>
+                                   </li>
+                                 );
+                               }
+                             })}
+                           </ul>
```

Comment:
You can simplify this code a lot and make the generated HTML more semantic by using a `<table>` instead of three `<ul>` lists. Right now there's these three different loops that have a lot of duplication between them, because you need to build the "table" column by column, so you need to loop over the diff three times.

If you use an actual `<table>` then you'd instead build it row by row. You'd have a single loop that would look something like this:

```jsx
<table>
  <thead>
    <tr>
      <td>Tag Name</td>
      <td>Old Value</td>
      <td>New Value</td>
    <tr>
  </thead>
  {_get(this.props, 'tagDiffs[0]') && Object.keys(_get(this.props, 'tagDiffs[0]')).map(tagName => {
    const tagChange = _get(this.props, 'tagDiffs[0]')[tagName];
    if (['changed', 'removed', 'added'].includes(tagChange.status)) {
      return (
        <tr>
          <td>{tagName}</td>
          <td>{tagChange.value || '—'}</td>
          <td>{tagChange.newValue || '—'}</td>
        </tr>
      )
    }
  })}
</table>

```